src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,512 @@
+   *     specified type {@code t}. May return an empty set.
+   */
+  public static Set<TypedOperation> getProducers(Type t) {
+    Set<TypedOperation> producerMethods = new LinkedHashSet<>();
+
+    // Search for methods that return the specified type in the specified classes.
+    for (String className : SPECIFIED_CLASSES) {
+      try {
+        Class<?> cls = Class.forName(className);
+        Type specifiedType = new NonParameterizedType(cls);
+        producerMethods.addAll(getProducers(t, specifiedType));
+      } catch (ClassNotFoundException e) {
+        throw new RandoopBug("Class not found: " + className);
+      }
+    }
+
+    // Search starting from the specified type.
+    producerMethods.addAll(getProducers(t, t));
+
+    return producerMethods;
+  }
+
+  /**
+   * Returns constructors/methods that return objects of the specified type.
+   *
+   * <p>Starting from {@code startingType}, examine all visible constructors/methods in it that
+   * return a type compatible with the specified type {@code t}. It recursively processes the inputs
+   * needed to execute these constructors and methods.
+   *
+   * @param t the return type of the resulting methods
+   * @param startingType the type from which to start the search
+   * @return a set of {@code TypedOperations} (constructors and methods) that return the specified
+   *     type {@code t}
+   */
+  private static Set<TypedOperation> getProducers(Type t, Type startingType) {
+    Set<Type> processed = new HashSet<>();
+    boolean isSearchingForTargetType = true;
+    List<TypedOperation> result = new ArrayList<>();
+    Set<Type> producerParameterTypes = new HashSet<>();
+    Queue<Type> workList = new ArrayDeque<>();
+    workList.add(startingType);
+
+    // Search for constructors/methods that can produce the specified type.
+    while (!workList.isEmpty()) {
+      Type currentType = workList.poll();
+
+      // Log the unspecified classes that are used in demand-driven input creation.
+      logIfUnspecified(currentType);
+
+      // Only consider the type if it is not a primitive type and if it hasn't already been
+      // processed.
+      if (!processed.contains(currentType) && !currentType.isNonreceiverType()) {
+        Class<?> currentClass = currentType.getRuntimeClass();
+        List<Executable> executableList = new ArrayList<>();
+
+        // Addi constructors if the current type is what we are looking for.
+        if (t.isAssignableFrom(currentType) && !Modifier.isAbstract(currentClass.getModifiers())) {
+          Collections.addAll(executableList, currentClass.getConstructors());
+        }
+
+        // Add methods that return the current type.
+        Collections.addAll(executableList, currentClass.getMethods());
+
+        // The first call checks for methods that return the specified type. Subsequent calls
+        // check for methods that return the current type.
```

Comment:
Why are subsequent calls different?  (More generally, I'm a bit confused about how this code operates.)